demo/src/checkout-pricing.js

```diff
@@ -15,24 +15,41 @@ function CheckoutPricingForm () {
   const [recurlyError, setRecurlyError] = useState(null);
   const [plan, setPlan] = useState('');
   const [quantity, setQuantity] = useState('');
+  const [coupon, setCoupon] = useState('');
+  const [giftCard, setGiftCard] = useState('');
+  const [currency, setCurrency] = useState('');
+  const [billingCountry, setBillingCountry] = useState('');
+  const [billingPostalCode, setBillingPostalCode] = useState('');
+  const [billingVatNumber, setBillingVatNumber] = useState('');
+  const [shippingCountry, setShippingCountry] = useState('');
+  const [shippingPostalCode, setShippingPostalCode] = useState('');
+  const [shippingVatNumber, setShippingVatNumber] = useState('');
   const [{ price, loading }, setPricing] = useCheckoutPricing({}, setRecurlyError);
+  const showPrice = price && price.now && !recurlyError
```

Comment:
I'm wondering if there's a better way to protect this output. Perhaps something we should note for improvement in recurly-js...

What is the scenario where this protection is required?

Reply:
My initial thought was that it's impossible to do this, as pricing will initially return undefined when the hook first runs. To give a little explanation:

When the hook first runs, it will immediately return the initial state, at which point `pricing` is undefined, causing `price` to return an empty object: https://github.com/recurly/react-recurly/blob/release/pricing/lib/use-checkout-pricing.js#L104.

After the pricingPromise resolves, state will update with the value returned from `recurly.Pricing.Checkout()` https://github.com/recurly/react-recurly/blob/release/pricing/lib/use-checkout-pricing.js#L42.

I think the best way to handle this would be to set the loading state to `true` by default, so consumers can check this flag to protect their render instead of doing these null checks. I'll add a commit to demonstrate what I mean that we can squash before merging.

Reply:
https://github.com/recurly/react-recurly/pull/25/commits/5ade2a170ed7d41e2f2cf21e552de577759a6ac7

Reply:
Short of writing a mocked out, static return value for the initial state of price here https://github.com/recurly/react-recurly/blob/release/pricing/lib/use-checkout-pricing.js#L43, I'm not sure how else to do this. Any thoughts?

Reply:
Love the use of loading, looks like a great recommendation that's easy to grok.

I think the best thing to get to here will be to construct a type as `CheckoutPricing.Price` in recurly-js. We can then import that to inform an initial state for this hook.